components/Favorites/FavoriteToggle.tsx

```diff
@@ -0,0 +1,83 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import StarBorderIcon from '@mui/icons-material/StarBorder'
+import StarIcon from '@mui/icons-material/Star'
+import styles from './FavoriteToggle.module.css'
+import { isItemFavorite, toggleFavoriteItem } from '../../utils/FavoriteItemUtils'
+import { CUSTOM_EVENTS } from '../../api/ApiTypes.d'
+
+interface Props {
+    item?: Item
+    className?: string
+    showLabel?: boolean
+    size?: 'small' | 'medium'
+}
+
+function FavoriteToggle({ item, className, showLabel = false, size = 'medium' }: Props) {
+    let [isFavorite, setIsFavorite] = useState(false)
+
+    useEffect(() => {
+        if (!item?.tag) {
+            setIsFavorite(false)
+            return
+        }
+        setIsFavorite(isItemFavorite(item.tag))
+    }, [item?.tag])
+
+    useEffect(() => {
+        function handleFavoritesUpdate(_event?: Event) {
+            if (!item?.tag) {
+                return
+            }
+            setIsFavorite(isItemFavorite(item.tag))
+        }
+        document.addEventListener(CUSTOM_EVENTS.FAVORITES_UPDATED, handleFavoritesUpdate)
+        return () => {
+            document.removeEventListener(CUSTOM_EVENTS.FAVORITES_UPDATED, handleFavoritesUpdate)
```

Comment:
dont rely on CUSTOM_EVENTS.

Also: Is it possible that the only way this is triggered, is through the "toggleFavoriteItem" function? The function is only used in this component, therefore this component is the only one that can even trigger this event. Why would you need to subscribe to it?